File: GeometryVisualization/src/core/figures/DisplayProperties.java
package core.figures;

import java.awt.Color;

public class DisplayProperties {
	private char style;
	private int thickness;
	private Color color;

	public DisplayProperties() {
		super();
	}

	public DisplayProperties(char style, int thickness, Color color) {
		super();
		this.style = style;
		this.thickness = thickness;
		this.color = color;
	}

	public char getStyle() {
		return style;
	}

	public void setStyle(char style) {
		this.style = style;
	}

	public int getThickness() {
		return thickness;
	}

	public void setThickness(int thickness) {
		this.thickness = thickness;
	}

	public Color getColor() {
		return color;
	}

	public void setColor(Color color) {
		this.color = color;
	}

	@Override
	public String toString() {
		return "([" + color.getRed() + "," + color.getGreen() + ","
				+ color.getBlue() + "], " + style + ", " + thickness + ")";
	}
}

File: cpp/graphical_simulation/Display_properties.cpp

#include "Display_properties.h"
#include <stdio.h>

Display_properties::Display_properties() {
}

Display_properties::Display_properties(string color, int tightness, string style) {
        this->color = color;
        this->thickness = tightness;
        this->style = style;
    }

string Display_properties::to_sim_format() {
        char bufffer[10];
        sprintf(bufffer, "%d", thickness);
        return color + "&" + style + "&" + string(bufffer);
    }

void Display_properties::set_properties(string color, int thickness, string style){
        this->color = color;
        this->thickness = thickness;
        this->style = style;
    }

